src/lib/network_pool/snark_pool_diff.ml

```diff
@@ -80,28 +80,43 @@ end)
   let apply (pool : Pool.t) (t : t Envelope.Incoming.t) :
       t Or_error.t Deferred.t =
     let open Deferred.Or_error.Let_syntax in
-    let data = Envelope.Incoming.data t in
+    let {Envelope.Incoming.data= diff; sender} = t in
+    let is_local = match sender with Local -> true | _ -> false in
     let to_or_error = function
-      | `Don't_rebroadcast ->
-          Or_error.error_string "Already in pool"
-      | `Rebroadcast ->
-          Ok data
+      | `Statement_not_referenced ->
+          Or_error.error_string "statement not referenced"
+      | `Added ->
+          Ok diff
     in
-    match data with
+    match diff with
     | Stable.V1.Add_solved_work (work, ({Priced_proof.proof; fee} as p)) -> (
+        let reject_and_log_if_local reason =
+          if is_local then
+            Logger.warn (Pool.get_logger pool) ~module_:__MODULE__
+              ~location:__LOC__
+              "Rejecting locally generated snark work $work, %s" reason
+              ~metadata:[("work", Work.Stable.V1.to_yojson work)] ;
```

Comment:
Could you use the `compact_json` function instead? that's what is logged in most of the places. Easier to track that way

Reply:
done